src/ray/raylet/node_manager.cc

```diff
@@ -460,6 +464,14 @@ void NodeManager::ProcessClientMessage(
 
       // Since some resources may have been released, we can try to dispatch more tasks.
       DispatchTasks();
+    } else {
+      // The client is a driver.
+      const std::shared_ptr<Worker> driver = worker_pool_.GetRegisteredDriver(client);
+      RAY_CHECK(driver);
+      auto driver_id = driver->GetAssignedTaskId();
```

Comment:
This seems like we're conflating "driver ID" with "task ID of the driver task". Is that correct? Is there a reason for doing that?

Reply:
I felt like it made sense to treat them as the same thing, since it seems like they pretty much serve the same purpose. What do you think?

Reply:
I agree they serve the same purpose.

I'm fine with unifying the IDs into the same thing, though I find it a little confusing to refer to it in some places as the "driver ID" and in some places as a "task ID". I think your approach is ok.

However, note that in the current PR, if we actually use it as a "task ID" (e.g., use it to look up something in the task table), that won't work because we're actually just passing in the "worker ID" (which I think is randomly generated) when we connect to the local scheduler.

Do you anticipate ever *using* the driver ID as a task ID in the node manager?

Reply:
Ah, I see. I think that's actually fixed since the most recent commit since I changed the driver's frontend to pass in `worker.current_task_id` instead of the worker ID. Do I understand that correctly?

Reply:
Oh ok, I think that's correct then. We confusingly set that ID for drivers in two places (both at the top and bottom of the snippet 

https://github.com/ray-project/ray/blob/05490b8cb9b321ed12e4e3e7df75545957a8766f/python/ray/worker.py#L2181-L2221